src/twisted/internet/test/test_unix.py

```diff
@@ -1014,3 +1017,121 @@ def connect(self, reactor, factory):
+
+        serverFactory = MyServerFactory()
+        serverConnMade: Deferred[AccumulatingProtocol] = Deferred()
+        serverFactory.protocolConnectionMade = serverConnMade
+        port = reactor.listenUNIX(self.path, serverFactory)
+        self.addCleanup(port.stopListening)
+
+        clientF = MyClientFactory()
+        clientConnMade: Deferred[Any] = Deferred()
+        clientF.protocolConnectionMade = clientConnMade
+        reactor.connectUNIX(self.path, clientF)
+
+        connsMade = gatherResults([serverConnMade, clientConnMade])
+
+        def connected(result: List[AccumulatingProtocol]) -> None:
+            serverProtocol, clientProtocol = result
+            callback(serverProtocol, clientProtocol)
+            assert serverProtocol.transport
+            assert clientProtocol.transport
+            serverProtocol.transport.loseConnection()
+            clientProtocol.transport.loseConnection()
+
+        return connsMade.addCallback(connected)
+
+    def test_tcpNoDelay(self) -> Deferred[None]:
+        """
+        The transport of a protocol connected with L{IReactorTCP.connectUNIX} or
+        L{IReactor.TCP.listenUNIX} can have L{ITCPTransport.getTcpNoDelay} or
+        L{ITCPTransport.setTcpNoDelay} called without error.
+        """
+
+        def check(
+            serverProtocol: AccumulatingProtocol, clientProtocol: AccumulatingProtocol
+        ) -> None:
+            for p in [serverProtocol, clientProtocol]:
+                assert interfaces.ITCPTransport.providedBy(p.transport)
+                transport = p.transport
+                self.assertEqual(transport.getTcpNoDelay(), 0)
+                transport.setTcpNoDelay(True)
+                self.assertEqual(transport.getTcpNoDelay(), 0)
+                transport.setTcpNoDelay(False)
+                self.assertEqual(transport.getTcpNoDelay(), 0)
+
+        return self._connectedClientAndServerTest(check)
+
+    def test_tcpKeepAlive(self) -> Deferred[None]:
+        """
+        The transport of a protocol connected with L{IReactorTCP.connectUNIX} or
+        L{IReactor.TCP.listenUNIX} can have its I{SO_KEEPALIVE} state inspected
+        and manipulated with L{ITCPTransport.getTcpKeepAlive} and
+        L{ITCPTransport.setTcpKeepAlive}.
+        """
+
+        def check(
+            serverProtocol: AccumulatingProtocol, clientProtocol: AccumulatingProtocol
+        ) -> None:
+            for p in [serverProtocol, clientProtocol]:
+                assert interfaces.ITCPTransport.providedBy(p.transport)
+                transport = p.transport
+                self.assertEqual(transport.getTcpKeepAlive(), False)
+                transport.setTcpKeepAlive(True)
+                self.assertEqual(transport.getTcpKeepAlive(), True)
+                transport.setTcpKeepAlive(False)
+                self.assertEqual(transport.getTcpKeepAlive(), False)
+
+        return self._connectedClientAndServerTest(check)
+
+    def testFailing(self) -> Deferred[Any]:
```

Comment:
Can we have a docstring for this test ?

```suggestion
    def test_Failing(self) -> Deferred[Any]:
```

I think that you can also write `async def` tests and yield a deferred and trial should just work 

it should make the tests easier to read

Reply:
Fixed.